### PR TITLE
Endpoint  to get cross place, stat_var stat for a given (or latest) date

### PR DESCRIPTION
**Pull request by shifucun**
Make a few new proto, go files and moved existing file there to modularize each file.

---
This change is [Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)

### CODE REVIEW

File: golden_response/staging/get_stat_multi/latest.json [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+    },
+    "Count_Person": {
+      "stat": {
+        "country/FRA": {
+          "date": "2020-06",
+          "value": 67081000,
+          "metadata": {
+            "importName": "WikipediaAndHumanCurated",
+            "provenanceUrl": "https://www.wikipedia.org",
+            "measurementMethod": "GoogleKGHumanCurated"
```
Comment (tjann) [inline golden_response/staging/get_stat_multi/latest.json:69]: We are ok exposing this?

**Review by tjann — APPROVED** [review]

File: golden_response/staging/get_stat_multi/latest.json (hunk 1 above)
Comment (shifucun) [reply golden_response/staging/get_stat_multi/latest.json:69]: @pradh shall we do WikipediaAndHumanCurated for mmethod as well?

**Review by pradh — CHANGES_REQUESTED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)* status: 0 of 18 files reviewed, 8 unresolved discussions (waiting on @beets, @pradh, @shifucun, and @tjann)

___
*[golden_response/staging/get_stat_multi/latest.json, line 69 at r2](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO025MxONa9P07DEtBV-r2-69:-MO2vR0p1VVgwf_6LAzE:b-v1a47a) ([raw file](https://github.com/datacommonsorg/mixer/blob/3708c270a901439d45d09c3cb2153c960aa893cc/golden_response/staging/get_stat_multi/latest.json#L69)):*
<details><summary><i>Previously, shifucun (Bo Xu) wrote…</i></summary><blockquote>

@pradh shall we do WikipediaAndHumanCurated for mmethod as well?
</blockquote></details>

it's a longer change. search is involved.

I'm also surprised this is the best ranked series for france?
___
*[proto/stat.proto, line 1 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2vjp_7mFkQwIlzWLG:-MO2vjp_7mFkQwIlzWLH:bboyojo) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L1)):*
> ```protobuf
> // Copyright 2019 Google LLC
> ```

Is this all content from mixer.proto?  Is there way to "branch" the file (for review and git history) ?
___
*[proto/stat.proto, line 22 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2w3OMEfsfqk0PoP-Q:-MO2w3OMEfsfqk0PoP-R:bje330u) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L22)):*
> ```protobuf
> 
> 
> message Metadata {
> ```

too broad?  StatMetadata?
___
*[proto/stat.proto, line 32 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2wOvoCh3csKjvLUt4:-MO2wOvoCh3csKjvLUt5:ba4keck) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L32)):*
> ```protobuf
> 
> message PointStat {
>   string date = 1;
> ```

document it is iso 8601?
___
*[proto/stat.proto, line 33 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2wMnv4Qbv7JMJ6bNK:-MO2wMnv4Qbv7JMJ6bNL:b13hmhr) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L33)):*
> ```protobuf
> message PointStat {
>   string date = 1;
>   double value = 2;
> ```

document?
___
*[proto/stat.proto, line 38 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2w0ZW5NVeJz-Wn9ME:-MO2w0ZW5NVeJz-Wn9MF:baeqb34) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L38)):*
> ```protobuf
> 
> message PlacePointStat {
>   map<string, PointStat> stat = 1;
> ```

Document key?
___
*[proto/stat.proto, line 271 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2vwBC3oD-hE4gGXeM:-MO2vwBC3oD-hE4gGXeN:baeqb34) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L271)):*
> ```protobuf
> 
> message GetStatMultiResponse {
>   map<string, PlacePointStat> data = 1;
> ```

Document key?
___
*[server/stat_bt_reader.go, line 1 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2wa2E2zyRx25WlvmZ:-MO2wa2E2zyRx25Wlvm_:b-mmo0aw) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/server/stat_bt_reader.go#L1)):*
> ```Go
> // Copyright 2020 Google LLC
> ```

Is this forked from stat.go? hard to review without branch :)

**Comment by beets** [comment]
+1 would be nice to review the post-move diffs

**Comment by shifucun** [comment]
Move refactored change out of this PR and addressed comments

**Review by shifucun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)* status: 0 of 16 files reviewed, 8 unresolved discussions (waiting on @beets, @pradh, @shifucun, and @tjann)

___
*[proto/stat.proto, line 1 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2vjp_7mFkQwIlzWLG:-MO3mmTl4iGiMAiprRaV:bw3l51o) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L1)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

Is this all content from mixer.proto?  Is there way to "branch" the file (for review and git history) ?
</blockquote></details>

this is a new file so can not really branch...
___
*[proto/stat.proto, line 22 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2w3OMEfsfqk0PoP-Q:-MO3ms0CCIGcK8H00Uz9:bb7f2mf) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L22)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

too broad?  StatMetadata?
</blockquote></details>

done
___
*[proto/stat.proto, line 32 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2wOvoCh3csKjvLUt4:-MO3mszy08awurPnMkbO:bb7f2mf) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L32)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

document it is iso 8601?
</blockquote></details>

done
___
*[proto/stat.proto, line 33 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2wMnv4Qbv7JMJ6bNK:-MO3mtuHBSgYGjF1NZkF:bb7f2mf) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L33)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

document?
</blockquote></details>

done
___
*[proto/stat.proto, line 271 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2vwBC3oD-hE4gGXeM:-MO3mulwA2nc6jhPJYMp:bb7f2mf) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L271)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

Document key?
</blockquote></details>

done
___
*[server/stat_bt_reader.go, line 1 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2wa2E2zyRx25WlvmZ:-MO3mwVO7kzs4IKhSJTx:b-5wpdjj) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/server/stat_bt_reader.go#L1)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

Is this forked from stat.go? hard to review without branch :)
</blockquote></details>

now only showing essential changes

**Review by shifucun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)* status: 0 of 16 files reviewed, 8 unresolved discussions (waiting on @beets, @pradh, @shifucun, and @tjann)

___
*[golden_response/staging/get_stat_multi/latest.json, line 69 at r2](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO025MxONa9P07DEtBV-r2-69:-MO3sGioE5nLIn_K3zKk:b-vqtsyv) ([raw file](https://github.com/datacommonsorg/mixer/blob/3708c270a901439d45d09c3cb2153c960aa893cc/golden_response/staging/get_stat_multi/latest.json#L69)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

it's a longer change. search is involved.

I'm also surprised this is the best ranked series for france?
</blockquote></details>

Updated the behavior so the highest ranked source data is returned

**Comment by shifucun** [comment]
This is ready for review now.

**Review by pradh — APPROVED** [review]
Thanks for the other changes to make this easy to review!

> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383#-:-MO3xVhO5icKHTR8ai-k:b-ps9wry)* status: 0 of 17 files reviewed, 2 unresolved discussions (waiting on @beets, @pradh, @shifucun, and @tjann)

**Review by beets — COMMENTED** [review]
> Reviewed 8 of 17 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)* status: 7 of 17 files reviewed, 3 unresolved discussions (waiting on @beets, @pradh, @shifucun, and @tjann)

___
*[server/stat_util.go, line 167 at r6](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO3xYWIBz9Qdx2EHpME:-MO3xYWIBz9Qdx2EHpMF:b3l6y22) ([raw file](https://github.com/datacommonsorg/mixer/blob/cc7301247e2a0b55143df81aca7047a5f585daf6/server/stat_util.go#L167)):*
> ```Go
> // When date is not given, it get the latest value from the highest ranked
> // source series.
> func getValueFromTopRank(in *ObsTimeSeries, date string) (float64, error) {
> ```

nit: maybe getValueFromTopSource or BestSource?

**Review by shifucun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)* status: 7 of 17 files reviewed, 2 unresolved discussions (waiting on @beets, @pradh, and @tjann)

___
*[proto/stat.proto, line 38 at r1](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO2w0ZW5NVeJz-Wn9ME:-MO3z86IE6lMNeBfXL8v:b-896fix) ([raw file](https://github.com/datacommonsorg/mixer/blob/4cc351bd27ee8d9e885b4f50c1e60ee73c87bb5c/proto/stat.proto#L38)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

Document key?
</blockquote></details>

Done.
___
*[server/stat_util.go, line 167 at r6](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO3xYWIBz9Qdx2EHpME:-MO3z-zGCAJVERqwN_Jy:bb6un1j) ([raw file](https://github.com/datacommonsorg/mixer/blob/cc7301247e2a0b55143df81aca7047a5f585daf6/server/stat_util.go#L167)):*
<details><summary><i>Previously, beets wrote…</i></summary><blockquote>

nit: maybe getValueFromTopSource or BestSource?
</blockquote></details>

Done

**Review by beets — APPROVED** [review]
nice!

> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383#-:-MO8p-93DNV49qC3SqkC:b-7m74ox)* status: 7 of 17 files reviewed, 2 unresolved discussions (waiting on @beets, @pradh, and @tjann)

___
*[server/stat_util.go, line 167 at r6](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO3xYWIBz9Qdx2EHpME:-MO8oiqV3U_XKeDvquux:b26di1n) ([raw file](https://github.com/datacommonsorg/mixer/blob/cc7301247e2a0b55143df81aca7047a5f585daf6/server/stat_util.go#L167)):*
<details><summary><i>Previously, shifucun (Bo Xu) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Looks like the other functions were changed except for this one.

**Review by shifucun — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/mixer/383)* status: 7 of 17 files reviewed, 2 unresolved discussions (waiting on @beets, @pradh, and @tjann)

___
*[server/stat_util.go, line 167 at r6](https://reviewable.io/reviews/datacommonsorg/mixer/383#-MO3xYWIBz9Qdx2EHpME:-MO9-dGb0OkaUgseX96j:b-6ptde6) ([raw file](https://github.com/datacommonsorg/mixer/blob/cc7301247e2a0b55143df81aca7047a5f585daf6/server/stat_util.go#L167)):*
<details><summary><i>Previously, beets wrote…</i></summary><blockquote>

Looks like the other functions were changed except for this one.
</blockquote></details>

updated as well

**Pull request merged**